### PR TITLE
Prettier Markdown (and thus better Documenter Support)

### PR DESCRIPTION
**Pull request by palday**
- Prettier Markdown
  -  This increases Documenter / older Julia StdLib support (which was finicky about alignment in tables)
  - It also makes things easier to read/preview in the REPL
- Added [a documentation page](https://juliastats.org/MixedModels.jl/previews/PR476/mime/) on using these methods
- Changed the older documentation to display the REPL output via hidden `DisplayAs.Text` lines.

(Merging this into the 3.x series for an immediate patch release, but I'll of course cherry-pick it over to 4.0-DEV)

### CODE REVIEW

File: test/markdown.jl [hunk 1]
```diff
@@ -31,57 +31,56 @@ include("modelcache.jl")
...
-|btype: shout|-2.1047525|0.258529|-8.14|<1e-15|         |         |
-|situ: self  |-1.0549790|0.210303|-5.02|<1e-6 |         |         |
+|              | Est.    |     SE |     z |      p | σ_subj | σ_item |
+|:------------ |:------- | ------:| -----:| ------:| ------:| ------:|
+| (Intercept)  | 0.1956  | 0.4052 |  0.48 | 0.6294 | 1.3398 | 0.4953 |
+| anger        | 0.0576  | 0.0168 |  3.43 | 0.0006 |        |        |
+| gender: M    | 0.3208  | 0.1913 |  1.68 | 0.0935 |        |        |
+| btype: scold | -1.0583 | 0.2568 | -4.12 | <1e-4  |        |        |
+| btype: shout | -2.1048 | 0.2585 | -8.14 | <1e-15 |        |        |
+| situ: self   | -1.0550 | 0.2103 | -5.02 | <1e-6  |        |        |  |
```
Comment (ararslan) [inline test/markdown.jl:67]:
```suggestion
| situ: self   | -1.0550 | 0.2103 | -5.02 | <1e-6  |        |        |
```
?

File: src/mdshow.jl [hunk 2]
```diff
@@ -93,91 +69,82 @@ function Base.show(io::IO, ::MIME"text/markdown", m::MixedModel)
...
+    push!(rows, ["`xtol_rel`", string(s.xtol_rel)])
+    push!(rows, ["`xtol_abs`", string(s.xtol_abs)])
+    push!(rows, ["`initial_step`", string(s.initial_step)])
+    push!(rows, ["`maxfeval`", string(s.maxfeval)])
+
+    push!(rows, ["**Result**",""])
+    push!(rows, ["Function evaluations", string(s.feval)])
+    push!(rows, ["Final parameter vector", "$(round.(s.final; digits=4))"])
+    push!(rows, ["Final objective value", "$(round.(s.fmin; digits=4))"])
+    push!(rows, ["Return code", "`$(s.returnvalue)`"])
```
Comment (ararslan) [inline src/mdshow.jl:141]:
```suggestion
    rows = [["", ""],

            ["**Initialization**", ""],
            ["Initial parameter vector", string(s.initial)],
            ["Initial objective value", string(s.finitial)],

            ["**Optimizer settings** ", ""],
            ["Optimizer (from NLopt)", "`$(s.optimizer)`"],
            ["`Lower bounds`", string(s.lowerbd)],
            ["`ftol_rel`", string(s.ftol_rel)],
            ["`ftol_abs`", string(s.ftol_abs)],
            ["`xtol_rel`", string(s.xtol_rel)],
            ["`xtol_abs`", string(s.xtol_abs)],
            ["`initial_step`", string(s.initial_step)],
            ["`maxfeval`", string(s.maxfeval)],

            ["**Result**",""],
            ["Function evaluations", string(s.feval)],
            ["Final parameter vector", "$(round.(s.final; digits=4))"],
            ["Final objective value", "$(round.(s.fmin; digits=4))"],
            ["Return code", "`$(s.returnvalue)`"]]
```
This is how I'd do it, FWIW, feel free to take it or leave it

Comment (palday) [reply src/mdshow.jl:141]: I was wondering whether you would catch some of the inefficiencies in all the `push!`ing, but a lot of it was just a convenient way to match the REPL-stuff that used `print` at each of those points.

**Review by ararslan — APPROVED** [review]
LGTM if CI agrees

File: src/mdshow.jl (hunk 2 above)
Comment (ararslan) [reply src/mdshow.jl:141]: I figured that was the case

**Pull request merged**